### PR TITLE
refactor(panel, flow-item, flow): removes flow-item, enhances panel to replace flow-item, updates flow to use panel

### PR DESCRIPTION
**Pull request by asangma**
**Related Issue:** #891

## Summary
* removes FlowItem
* enhances Panel
  - adds back button
  - adds slot for dropdown menu (todo refactor to use CalciteDropdown once it's updated to use Popper)
  - allows both multiple actions at end and dropdown menu
* refactors Flow to use Panel
* demo updates
* e2e updates
* readme updates

TODO 
- [x] footer-content slot
  - this will override footer-actions similar to how header-content works in header
- [x] footer-actions slot(s)
- [x] Probably some clean up

### CODE REVIEW

**Comment by driskull** [comment]
This is looking good @asangma! Let me know once the `TODO` stuff is in and I'll do a thorough review.

**Comment by asangma** [comment]
@asangma 
Will do.

**Comment by asangma** [comment]
Ready for review.

File: src/components/calcite-panel/calcite-panel.tsx [hunk 1]
```diff
@@ -180,45 +342,126 @@ export class CalcitePanel {
...
+        <slot name={SLOTS.footerActions}></slot>
+      </footer>
+    ) : null;
+  }
+
+  renderFooter(): VNode {
+    const footerContentNode = this.renderFooterSlottedContent()
+    const footerActionsNode = this.renderFooterActions();
+
+    return footerContentNode || footerActionsNode;
```
Comment (driskull) [inline src/components/calcite-panel/calcite-panel.tsx:462]: Lets just simplify it to `return this.renderFooterSlottedContent() || this.renderFooterActions();`

File: src/components/calcite-panel/resources.ts [hunk 2]
```diff
@@ -1,26 +1,40 @@
 export const CSS = {
+  backButton: "back-button",
   container: "container",
   header: "header",
-  headerLeadingContent: "header-leading-content",
+  heading: "heading",
+  summary: "summary",
   headerContent: "header-content",
-  headerTrailingContent: "header-trailing-content",
+  headerActionsEnd: "header-actions-end",
```
Comment (driskull) [inline src/components/calcite-panel/resources.ts:8]: I wonder if we want a class for `header-actions` and then modifiers `header-actions--end`, `header-actions--start`?

**Review by driskull — APPROVED** [review]
Looks 💣  @asangma. Pulled down and checked it out. Seems to work good 👍

File: src/components/calcite-panel/resources.ts (hunk 2 above)
Comment (asangma) [reply src/components/calcite-panel/resources.ts:8]: Ah yeah...good call.

File: src/components/calcite-panel/calcite-panel.tsx (hunk 1 above)
Comment (asangma) [reply src/components/calcite-panel/calcite-panel.tsx:462]: Copy that.

Comment (asangma) [reply src/components/calcite-panel/calcite-panel.tsx]: Updated. Let me know if I went too far on simplifying.

File: src/components/calcite-panel/resources.ts (hunk 2 above)
Comment (asangma) [reply src/components/calcite-panel/resources.ts]: Thanks @driskull. Updated.

**Review by driskull — APPROVED** [review]
Looks good!

**Comment by asangma** [comment]
Thanks for jumping on the review @driskull. 
I updated the e2e with a host of tests.

**Comment by asangma** [comment]
Discovered some scroll and sticky positioning conflicts with Panel, ShellPanel, and Flow. Please hold off on reviewing.

**Comment by asangma** [comment]
@driskull @jcfranco @macandcheese All right y'all. I think this one is ready for your review.

File: src/components/calcite-panel/calcite-panel.scss [hunk 3]
```diff
@@ -94,20 +150,25 @@ calcite-scrim {
...
+  bottom: 0;
+  width: 100%;
 }
 
 .calcite--rtl {
-  .header-leading-content + .header-content {
-    padding-right: var(--calcite-spacing-half);
+  .header-content {
+    margin-left: auto;
+    margin-right: 0;
```
Comment (driskull) [inline src/components/calcite-panel/calcite-panel.scss:168]: should this use unset too?

File: src/components/calcite-panel/calcite-panel.tsx [hunk 4]
```diff
@@ -180,45 +362,130 @@ export class CalcitePanel {
...
+    return (
+      <calcite-popover
+        disablePointer={true}
+        flipPlacements={["bottom-end", "top-end"]}
+        offsetDistance={0}
+        onKeyDown={this.menuActionsKeydown}
+        open={menuOpen}
+        placement="bottom-end"
+        referenceElement={menuButtonEl}
+        theme={getElementTheme(el)}
```
Comment (driskull) [inline src/components/calcite-panel/calcite-panel.tsx:391]: I'm not sure we need to pass the theme anymore here. Now that we're sharing theme styling? Is that correct? We can probably remove a lot of places where we say `theme={...`

File: src/components/calcite-panel/calcite-panel.tsx [hunk 5]
```diff
@@ -180,45 +362,130 @@ export class CalcitePanel {
...
         {dismissibleNode}
       </div>
+    ) : null;
+  }
+
+  renderMenuItems(): VNode {
+    const { el, menuOpen, menuButtonEl } = this;
+
+    return (
+      <calcite-popover
```
Comment (driskull) [inline src/components/calcite-panel/calcite-panel.tsx:383]: We'll use `calcite-dropdown` for this at some point right? Would we still place an action inside of the dropdown to trigger it?

**Review by driskull — APPROVED** [review]
Looks good @asangma 🎉 🌮

File: src/components/calcite-panel/calcite-panel.scss (hunk 3 above)
Comment (asangma) [reply src/components/calcite-panel/calcite-panel.scss:168]: Oooh...I'll check that out.

File: src/components/calcite-panel/calcite-panel.tsx (hunk 4 above)
Comment (asangma) [reply src/components/calcite-panel/calcite-panel.tsx:391]: I can remove it.

File: src/components/calcite-panel/calcite-panel.tsx (hunk 5 above)
Comment (asangma) [reply src/components/calcite-panel/calcite-panel.tsx:383]: That's what I'm hoping for. I think we'd still use an Action for the trigger.

File: src/demos/shell/demo-app-full-window.html [hunk 6]
```diff
@@ -48,8 +48,8 @@
     <main>
       <div class="shell-container">
         <calcite-shell>
-          <calcite-shell-panel slot="primary-panel" position="start" detached>
-            <calcite-panel
+          <calcite-shell-panel slot="primary-panel" position="start" >
+            <!-- <calcite-panel
```
Comment (jcfranco) [inline src/demos/shell/demo-app-full-window.html:52]: Please remove commented out code if no longer applicable. This applies to other lines as well.

File: src/demos/shell/demo-app-full-window.html [hunk 7]
```diff
@@ -455,33 +460,33 @@ <h3 class="heading" slot="header-content">WE CARE A LOT</h3>
             : tipManager.setAttribute("hidden", "");
         });
 
-        const actionPad = document.getElementById("action-pad");
-        const actionPadButton = document.getElementById("action-pad-button");
-        actionPadButton.addEventListener("click", function () {
-          actionPad.hasAttribute("hidden") ? actionPad.removeAttribute("hidden") : actionPad.setAttribute("hidden", "");
+        // const actionPad = document.getElementById("action-pad");
```
Comment (jcfranco) [inline src/demos/shell/demo-app-full-window.html:463]: Is this no longer needed?

**Review by jcfranco — APPROVED** [review]
Other than a few comments, this lewks gewd! 💪😎 👍

File: src/demos/shell/demo-app-full-window.html (hunk 6 above)
Comment (asangma) [reply src/demos/shell/demo-app-full-window.html:52]: 👍

File: src/demos/shell/demo-app-full-window.html (hunk 7 above)
Comment (asangma) [reply src/demos/shell/demo-app-full-window.html:463]: Will clean it up.

File: src/components/calcite-panel/calcite-panel.tsx [hunk 8]
```diff
@@ -180,45 +362,130 @@ export class CalcitePanel {
...
+    const menuLabel = menuOpen ? closeLabel : openLabel;
+
+    return (
+      <calcite-action
+        aria-label={menuLabel}
+        class={CSS.menuButton}
+        icon={ICONS.menu}
+        onClick={this.toggleMenuOpen}
+        onKeyDown={this.menuButtonKeyDown}
+        ref={(menuButtonEl): HTMLCalciteActionElement => (this.menuButtonEl = menuButtonEl)}
```
[the remark this answers is not in the document]
Comment (asangma) [reply src/components/calcite-panel/calcite-panel.tsx:414]: Copy that.

[the remark this answers is not in the document]
Comment (asangma) [reply src/components/calcite-panel/calcite-panel.tsx:414]: Actually @jcfranco...I don't know how to do that. 😆

[the remark this answers is not in the document]
Comment (jcfranco) [reply src/components/calcite-panel/calcite-panel.tsx:414]:
```ts
renderMenuButton(): VNode {
  //...
  return <calcite-action
     //...
     ref={this.setMenuButton}
     text={menuLabel}
    />
}

setMenuButtonEl(el: HTMLCalciteActionElement): void {
  this.menuButtonEl = menuButtonEl;
}
```

[the remark this answers is not in the document]
Comment (asangma) [reply src/components/calcite-panel/calcite-panel.tsx:414]: 🙏

File: src/demos/shell/demo-app-full-window.html (hunk 7 above)
Comment (asangma) [reply src/demos/shell/demo-app-full-window.html]: Cleaned up.

File: src/components/calcite-panel/calcite-panel.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (asangma) [reply src/components/calcite-panel/calcite-panel.tsx]: Thanks @jcfranco! Updated with your recommended changes.

**Comment by asangma** [comment]
Unless there's anything else, I will murgle.

**Pull request merged**